src/evict/evict_lru.c

```diff
@@ -574,52 +558,70 @@ __evict_pass(WT_SESSION_IMPL *session)
 		 * workers, it must service the urgent queue in case all
 		 * application threads are busy.
 		 */
-		if (cache->evict_empty_score < WT_EVICT_EMPTY_SCORE_CUTOFF ||
+		if (cache->evict_empty_score < WT_EVICT_SCORE_CUTOFF ||
 		    (!WT_EVICT_HAS_WORKERS(session) &&
 		    !__evict_queue_empty(cache->evict_urgent_queue)))
 			WT_RET(__evict_lru_pages(session, true));
 
+		if (cache->pass_intr != 0)
```

Comment:
I'm not inclined to chase this now - I think it's more likely to be relevant in different workloads. I mentioned it because I knew I'd completely forget otherwise.